Objects/Converters/ConverterCSI/ConverterCSIShared/ConverterCSIUtils.cs

```diff
@@ -27,18 +27,45 @@ public string ModelUnits()
       return _modelUnits;
     }
 
+#if ETABS || SAFE
+
+    _modelUnits = GetModelUnitsFromETABS();
+
+#else
+
+    _modelUnits = GetModelUnitsFromSAP();
+
+#endif
+
+    return _modelUnits;
+  }
+
+  public string GetModelUnitsFromETABS()
+  {
     eForce forceUnits = eForce.NotApplicable;
     eLength lengthUnits = eLength.NotApplicable;
     eTemperature temperatureUnits = eTemperature.NotApplicable;
+    // GetPresentUnits_2() works for ETABS and SAFE
     _ = Model.GetPresentUnits_2(ref forceUnits, ref lengthUnits, ref temperatureUnits);
 
     if (lengthUnits == eLength.NotApplicable)
     {
       throw new SpeckleException("Unable to retreive valid length units from the ETABS document");
     }
 
-    _modelUnits = lengthUnits.ToString();
-    return _modelUnits;
+    return lengthUnits.ToString();
+  }
+
+  public string GetModelUnitsFromSAP()
+  {
+    // GetPresentUnits() works for SAP 2000 and CSIBridge
+    var units = Model.GetPresentUnits();
+    if (units != 0)
+    {
+      string[] unitsCat = units.ToString().Split('_');
```

Comment:
Are we sure its safe to assume that `GetPresentUnits` will always return a value that has an `'_'`?

Is this documented in their API?

Reply:
From the SAP 2000 documentation:

<img width="605" alt="image" src="https://github.com/user-attachments/assets/f9e1d117-8222-493a-a101-ccef5cad4951">

Hence index positions for 0, 1 and 2 being force, distance and temp respectively. In SAP interface these enums seem well enforced (I, as the user, can't define something funny):

<img width="329" alt="image" src="https://github.com/user-attachments/assets/aa78ca76-e638-4eec-bc9d-13668bc5006a">

Reply:
Looks good

Reply:
Just got my hands on the documentation, identical to SAP 2000 😄